spark/src/main/java/org/apache/iceberg/spark/data/SparkValueReaders.java

```diff
@@ -96,6 +101,24 @@ public UTF8String read(Decoder decoder, Object reuse) throws IOException {
     }
   }
 
+  private static class EnumReader implements ValueReader<UTF8String> {
+    private final byte[][] symbols;
+
+    private EnumReader(List<String> symbols) {
+      this.symbols = new byte[symbols.size()][];
+      for (int i = 0; i < this.symbols.length; i += 1) {
+        this.symbols[i] = symbols.get(i).getBytes(StandardCharsets.UTF_8);
+      }
+    }
+
+    @Override
+    public UTF8String read(Decoder decoder, Object ignore) throws IOException {
+      int index = decoder.readEnum();
+      return UTF8String.fromBytes(symbols[index], 0, symbols[index].length);
```

Comment:
Why instantiate a new `UTF8String` each time instead of reusing the same one for each symbol?